aws-datastore/src/main/java/com/amplifyframework/datastore/syncengine/ModelSyncMetricsAccumulator.java

```diff
@@ -0,0 +1,73 @@
+/*
+ * Copyright 2020 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *  http://aws.amazon.com/apache2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+package com.amplifyframework.datastore.syncengine;
+
+import com.amplifyframework.core.Amplify;
+import com.amplifyframework.core.model.Model;
+import com.amplifyframework.datastore.DataStoreItemChange;
+import com.amplifyframework.datastore.events.ModelSyncedEvent;
+import com.amplifyframework.datastore.storage.LocalStorageAdapter;
+import com.amplifyframework.datastore.storage.StorageItemChange;
+import com.amplifyframework.logging.Logger;
+
+import java.util.Map;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.concurrent.atomic.AtomicInteger;
+
+/**
+ * Class that encapsulates the logic of keeping track of sync metrics
+ * by operation type for a given model.
+ */
+final class ModelSyncMetricsAccumulator {
+    private static final Logger LOG = Amplify.Logging.forNamespace("amplify:aws-datastore");
+    private final Map<String, AtomicInteger> syncMetrics;
```

Comment:
Does it make sense for the key to be the constrained type, `DataStoreItemChange.TYPE`, instead of just `String`?

Reply:
You know...I tried to do this and couldn't get it to work. Finally just realized why: there is a `StorageItemChange.Type` and a `DataStoreItemChange.Type`. When I initially tried this, I was using `StorageItemChange.Type` in the key declaration, and then trying to retrieve by passing in a variable of `DataStoreItemChange.Type`. Just brilliant 🤣